packages/react-router-dom/index.tsx

```diff
@@ -390,7 +390,7 @@ export const Link = React.forwardRef<HTMLAnchorElement, LinkProps>(
       <a
         {...rest}
         href={href}
-        onClick={handleClick}
+        onClick={reloadDocument ? onClick : handleClick}
```

Comment:
No longer attach our own listener if `reloadDocument==false`, but ensure we attach _their_ listener if provided so they can `preventDefault` on `<Link reloadDodument={true}`>`